firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -32,52 +32,42 @@
 import java.net.HttpURLConnection;
 import java.util.Random;
 import java.util.Set;
-import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import org.json.JSONException;
 import org.json.JSONObject;
 
-public class ConfigAutoFetch {
+public class ConfigAutoFetch implements Runnable {
 
   private static final int FETCH_RETRY = 3;
 
   @GuardedBy("this")
   private final Set<ConfigUpdateListener> eventListeners;
 
-  @GuardedBy("this")
   private final HttpURLConnection httpURLConnection;
 
   private final ConfigFetchHandler configFetchHandler;
   private final ConfigUpdateListener retryCallback;
   private final ScheduledExecutorService scheduledExecutorService;
   private final Random random;
-  private final Executor executor;
 
   public ConfigAutoFetch(
       HttpURLConnection httpURLConnection,
       ConfigFetchHandler configFetchHandler,
       Set<ConfigUpdateListener> eventListeners,
-      ConfigUpdateListener retryCallback,
-      Executor executor,
-      ScheduledExecutorService scheduledExecutorService) {
+      ConfigUpdateListener retryCallback) {
     this.httpURLConnection = httpURLConnection;
     this.configFetchHandler = configFetchHandler;
     this.eventListeners = eventListeners;
     this.retryCallback = retryCallback;
-    this.scheduledExecutorService = scheduledExecutorService;
+    this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
It's usually a good idea to receive the executor as a parameter to make testing easier, what's the rationale behind the change?

Reply:
The idea was so that each autoFetch thread would get an executor to handle fetches. That way when the autoFetch thread dies/ends we can also stop the respective scheduled executor to close the fetch threads and so they don't stay running.